internal/pkg/agent/application/upgrade/step_mark.go

```diff
@@ -38,6 +38,58 @@ type UpdateMarker struct {
 	Action *fleetapi.ActionUpgrade `json:"action" yaml:"action"`
 }
 
+// MarkerActionUpgrade adapter struct compatible with pre 8.3 version of the marker file format
+type MarkerActionUpgrade struct {
+	ActionID   string `yaml:"id"`
```

Comment:
I think a part of the update failed because I changed the serialization of the upgrade action with
https://github.com/elastic/elastic-agent/blob/7db7406cc04d6dd8f2d47f8a37e299da4598056b/internal/pkg/fleetapi/action.go#L193-L194

I tried to make it more inline in [fleet-server/model/schema.json](https://github.com/elastic/fleet-server/blob/main/model/schema.json) (though fleet-server will remap `action_id` to `id` when giving an agent its actions on checkin).
Should we revisit that?